File: ruby/cukes/step_definitions.rb
# encoding: UTF-8

require_relative 'actionwords'
World(Actionwords)

Given /^I start the coffee machine "(.*)"$/ do |lang|
  i_start_the_coffee_machine(lang)
end

When /^I shutdown the coffee machine$/ do
  i_shutdown_the_coffee_machine
end

Then /^message "(.*)" should be displayed$/ do |message|
  message_message_should_be_displayed(message)
end

Then /^coffee should be served$/ do
  coffee_should_be_served
end

Then /^coffee should not be served$/ do
  coffee_should_not_be_served
end

When /^I take a coffee$/ do
  i_take_a_coffee
end

Given /^I empty the coffee grounds$/ do
  i_empty_the_coffee_grounds
end

And /^I fill the beans tank$/ do
  i_fill_the_beans_tank
end

When /^I fill the water tank$/ do
  i_fill_the_water_tank
end

When /^I take "(.*)" coffees$/ do |coffee_number|
  i_take_coffee_number_coffees(coffee_number)
end

Given /^the coffee machine is started$/ do
  the_coffee_machine_is_started
end

When /^fifty coffees have been taken without filling the tank$/ do
  fifty_coffees_have_been_taken_without_filling_the_tank
end

When /^thirty eight coffees are taken without filling beans$/ do
  thirty_eight_coffees_are_taken_without_filling_beans
end

File: specflow/features/StepDefinitions.cs
namespace Hiptest.Publisher.Samples {

    [Binding]
    public class StepDefinitions {

        [Given(@"^I start the coffee machine \"(.*)\"$")]
        public void IStartTheCoffeeMachine(null lang) {
            IStartTheCoffeeMachine(null lang);
        }


        [Given(@"^I shutdown the coffee machine$")]
        public void IShutdownTheCoffeeMachine() {
            IShutdownTheCoffeeMachine();
        }


        [Given(@"^message \"(.*)\" should be displayed$")]
        public void MessageMessageShouldBeDisplayed(string message) {
            MessageMessageShouldBeDisplayed(string message);
        }


        [Given(@"^coffee should be served$")]
        public void CoffeeShouldBeServed() {
            CoffeeShouldBeServed();
        }


        [Given(@"^coffee should not be served$")]
        public void CoffeeShouldNotBeServed() {
            CoffeeShouldNotBeServed();
        }


        [Given(@"^I take a coffee$")]
        public void ITakeACoffee() {
            ITakeACoffee();
        }


        [Given(@"^I empty the coffee grounds$")]
        public void IEmptyTheCoffeeGrounds() {
            IEmptyTheCoffeeGrounds();
        }


        [Given(@"^I fill the beans tank$")]
        public void IFillTheBeansTank() {
            IFillTheBeansTank();
        }


        [Given(@"^I fill the water tank$")]
        public void IFillTheWaterTank() {
            IFillTheWaterTank();
        }


        [Given(@"^I take \"(.*)\" coffees$")]
        public void ITakeCoffeeNumberCoffees(int coffeeNumber) {
            ITakeCoffeeNumberCoffees(int coffeeNumber);
        }


        [Given(@"^the coffee machine is started$")]
        public void TheCoffeeMachineIsStarted() {
            TheCoffeeMachineIsStarted();
        }


        [Given(@"^fifty coffees have been taken without filling the tank$")]
        public void FiftyCoffeesHaveBeenTakenWithoutFillingTheTank() {
            FiftyCoffeesHaveBeenTakenWithoutFillingTheTank();
        }


        [Given(@"^thirty eight coffees are taken without filling beans$")]
        public void ThirtyEightCoffeesAreTakenWithoutFillingBeans() {
            ThirtyEightCoffeesAreTakenWithoutFillingBeans();
        }
    }
}